libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -3363,16 +3375,25 @@ void IRGeneratorForStatements::revertWithError(
 	vector<ASTPointer<Expression const>> const& _errorArguments
 )
 {
+	bool needsAllocation = true;
+	if (auto size = staticEncodingSize(_parameterTypes))
+		needsAllocation = *size > CompilerUtils::generalPurposeMemoryStart;
 	Whiskers templ(R"({
+		<?needsAllocation>
 		let <pos> := <allocateUnbounded>()
+		<!needsAllocation>
+		let <pos> := 0
```

Comment:
I'm surprised this change is reflected nowhere in the tests, in other words, I assume `allocateUnbounded` always returns 0 in our tests, which would mean that some coverage is missing.

Also, if this makes things cheaper, wouldn't this also be reflected in the gas cost test? Or I guess not, if `pos` was/is always 0?

Reply:
Why would you assume that? Then the change would just be nonsense :-D.
``allocateUnbounded`` never returns zero.
This is most definitely reflected in our tests - the behaviour just doesn't change (it can't, otherwise, it'd be a bug...) and we don't have tests involving eligible errors that consume sufficiently much gas, s.t. this change should show up in the gas statistics. We generally don't have verbatim tests on complex bytecode, since those would just break all the time, but this is covered in at least 50, probably more semantics tests.